### PR TITLE
README and CONTRIBUTING

### PR DESCRIPTION
**Pull request by tarebyte**

### CODE REVIEW

**Comment by tarebyte** [comment]
@johndbritton this is ready for your :eyes:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+[fork]: https://github.com/education/classroom/fork
+[pr]: https://github.com/education/classroom/compare
+[style]: https://github.com/bbatsov/ruby-style-guide
+
+Hi there! We're thrilled that you'd like to contribute to this project. Your help is essential for keeping it great.
+
+## Submitting a pull request
+
+0. [Fork][fork] and clone the repository
+0. Configure and install the dependencies: (See the [README](README.md) for more details)
```
Comment (johndbritton) [inline CONTRIBUTING.md]: Maybe link to `https://github.com/education/classroom/blob/readme/README.md#get-started`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+0. Push to your fork and [submit a pull request][pr]
+0. Pat your self on the back and wait for your pull request to be reviewed and merged.
+
+Here are a few things you can do that will increase the likelihood of your pull request being accepted:
+
+- Follow the [style guide][style].
+- Write tests.
+- Keep your change as focused as possible. If there are multiple changes you would like to make that are not dependent upon each other, consider submitting them as separate pull requests.
+- Write a [good commit message](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+
```
Comment (johndbritton) [inline CONTRIBUTING.md]: Could use a section on reporting issues, what makes a good report, and searching for duplicates.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,91 @@
-# Classroom
+# Classroom for GitHub
 [![Build Status](https://travis-ci.org/education/classroom.svg?branch=master)](https://travis-ci.org/education/classroom)
+
+> Your course assignments on GitHub
```
Comment (johndbritton) [inline README.md]: `## Your course assignments on GitHub`

File: README.md [hunk 4]
```diff
@@ -1,2 +1,91 @@
...
+
+Classroom for GitHub is a [Ruby on Rails](http://rubyonrails.org/) application designed to automate repository creation and access control, making it easy to distribute starter code and collect assignments on GitHub
+
+![Classroom for GitHub screenshot](https://cloud.githubusercontent.com/assets/123345/9819714/95b26b9c-58a7-11e5-99e9-d65996d53687.png)
+
+## How it works
+
+Classroom for Github automates repository creation and access control, making it easy to distribute starter code and collect assignments from students.
```
Comment (johndbritton) [inline README.md]: Repetitive, you can strike this whole line.

File: CONTRIBUTING.md (hunk 1 above)
Comment (johndbritton) [inline CONTRIBUTING.md]: or `https://github.com/education/classroom/blob/readme/README.md#hacking-on-classroom-for-github`

File: README.md [hunk 5]
```diff
@@ -1,2 +1,91 @@
...
+deploy
+
+## Hacking on Classroom for GitHub
+
+### Get started
+New to Ruby? No worries! You can follow these instructions to install a local server.
+
+#### Installing a Local Server
+
+First things first, you'll need to install Ruby 2.2.3. I recommend using the excellent [rbenv](https://github.com/sstephenson/rbenv),
```
Comment (johndbritton) [inline README.md]: We > I

File: README.md [hunk 6]
```diff
@@ -1,2 +1,91 @@
...
+```
+
+Now, let's install the gems from the `Gemfile` ("Gems" are synonymous with libraries in other
+languages).
+
+```bash
+gem install bundler && rbenv rehash
+```
+
+### Setup the Classroom for GitHub
```
Comment (johndbritton) [inline README.md]: `### Set up Classroom for GitHub` or `### Set up the application`

File: README.md [hunk 7]
```diff
@@ -1,2 +1,91 @@
...
+Once bundler is installed go ahead and run the setup script
```
Comment (johndbritton) [inline README.md]: `setup` should be quoted.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,91 @@
...
+```
+script/setup
+```
+
+### Development environment variables
+We use [dotenv](https://github.com/bkeepers/dotenv) for setting our environment variables. These values need to filled in in you `.env` file (which was created by the `setup` script.
```
Comment (johndbritton) [inline README.md]: > These values must be present in your `.env` file (created by `script/setup`).

File: README.md [hunk 9]
```diff
@@ -1,2 +1,91 @@
...
+
+ENV Variable | Description |
+:-------------------|:-----------------|
+`GITHUB_CLIENT_ID`| the GitHub.com Application Client ID..
+`GITHUB_CLIENT_SECRET`| the GitHub.com Application Client Secret..
```
Comment (johndbritton) [inline README.md]: ~~.com~~

File: README.md [hunk 10]
```diff
@@ -1,2 +1,91 @@
...
+`NON_STAFF_GITHUB_ADMIN_IDS` | GitHub IDs of non GitHub staff members that have staff access.
```
Comment (johndbritton) [inline README.md]: GitHub `user_id`s of users to be granted staff level access.

File: README.md [hunk 11]
```diff
@@ -1,2 +1,91 @@
...
+
+### Testing environment variables
+
+ENV Variable | Description |
+:-------------------|:-----------------|
+`CLASSROOM_OWNER_ID` | The GitHub user ID of an organization admin.
```
Comment (johndbritton) [inline README.md]: `user_id`

File: README.md [hunk 12]
```diff
@@ -1,2 +1,91 @@
...
+`CLASSROOM_OWNER_GITHUB_TOKEN` | The [Personal Access Token](https://github.com/blog/1509-personal-api-tokens) for the owner ID
```
Comment (johndbritton) [inline README.md]: This description is unclear, I don't know what this does.

File: README.md [hunk 13]
```diff
@@ -1,2 +1,91 @@
...
+### Development environment variables
+We use [dotenv](https://github.com/bkeepers/dotenv) for setting our environment variables. These values need to filled in in you `.env` file (which was created by the `setup` script.
+
+ENV Variable | Description |
+:-------------------|:-----------------|
+`GITHUB_CLIENT_ID`| the GitHub.com Application Client ID..
+`GITHUB_CLIENT_SECRET`| the GitHub.com Application Client Secret..
+`NON_STAFF_GITHUB_ADMIN_IDS` | GitHub IDs of non GitHub staff members that have staff access.
+
+### Testing environment variables
```
Comment (johndbritton) [inline README.md]: Maybe a paragraph about what these variables are for and how they're used. It seems like I can set these to override actual logged in users in development mode.

Comment (johndbritton) [inline README.md]: Also maybe update these vars to have a `TEST_` prefix for clarity?

File: README.md [hunk 14]
```diff
@@ -1,2 +1,91 @@
...
+And that's it! You should have a working instance of Classroom for GitHub located [here](http://localhost:3000)
+
+## Deployment
+
+We strongly encourage you to use [https://classroom.github.com](https://classroom.github.com), but if you would like your own version Classroom for GitHub can be easily deployed to Heroku.
+
+[![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy)
+
+## Contributing
+We'd love to have you participate. Please check out [contributing guidelines](CONTRIBUTING.md)
```
Comment (johndbritton) [inline README.md]: Missing period.

File: README.md [hunk 15]
```diff
@@ -1,2 +1,91 @@
...
+
```
Comment (johndbritton) [inline README.md]: > `### Contributors`
> Classroom is developed by the following [contributors](https://github.com/education/classroom/graphs/contributors).

**Comment by johndbritton** [comment]
Made a bunch of comments, but they're mostly superficial. :shipit: once they're resolved.

File: README.md (hunk 13 above)
Comment (tarebyte) [inline README.md]: > Maybe a paragraph about what these variables are for and how they're used.

:+1:


> It seems like I can set these to override actual logged in users in development mode.

Are you referring to `NON_STAFF_GITHUB_ADMIN_IDS`?

Comment (johndbritton) [inline README.md]: I'm referring to all the "Test" environment variables.

**Pull request merged**